src/main/java/com/yugabyte/sample/apps/MultiTableMultiIndexInserts.java

```diff
@@ -0,0 +1,382 @@
+    statement.execute(String.format("CREATE TABLE %s ( m_id uuid NOT NULL,a_id text, m_key text NOT NULL,m_value text);", getTableName("m")));
+    statement.execute(String.format("CREATE TABLE %s (id integer primary key, v character varying(30) NOT NULL);", getTableName("mv")));
+    LOG.info("Created all the tables");
+
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (t);", getTableName("a1_t_idx"), getTableName("a1")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (f_id_1); ", getTableName("a1_f_id_1_idx"), getTableName("a1")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (f_id_2);", getTableName("a1_f_id_2_idx"), getTableName("a1")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (id);", getTableName("a1a_id_idx"), getTableName("a1a")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (t);", getTableName("a1a_t_idx"), getTableName("a1a")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (a2_id);", getTableName("a1a_a2_id_idx"), getTableName("a1a")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (t);", getTableName("a2_t_idx"), getTableName("a2")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (t);", getTableName("a3_t"), getTableName("a3")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (t);", getTableName("a3a_t_idx"), getTableName("a3a")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (a2_id);", getTableName("a3a_a2_id_idx"), getTableName("a3a")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (a3_id);", getTableName("a3a_a3_id_idx"), getTableName("a3a")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (t);", getTableName("d_t_idx"), getTableName("d")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (mv_id);", getTableName("d_mv_idx"), getTableName("d")));
+    connection.createStatement().execute(String.format("CREATE INDEX %s ON %s (m_id);", getTableName("m_idx"), getTableName("m")));
+    LOG.info("Created all the indexes");
+  }
+
+  public long doRead() {
+    return 1L;
+  }
+
+  long getNextLong(int threadIdx) {
+    long a = nextLong.get(threadIdx);
+    long b = nextLongMax.get(threadIdx);
+    if (a > b) {
+      a = atomicLong.getAndAdd(kAtomicDelta);
+      b = a + kAtomicDelta - 1L;
+      nextLongMax.put(threadIdx, b);
+    }
+
+    nextLong.put(threadIdx, a + 1L);
+    return a;
+  }
+
+  int getNextInt(int threadIdx) {
+    int a = nextInt.get(threadIdx);
+    int b = nextIntMax.get(threadIdx);
+    if (a > b) {
+      a = atomicInt.getAndAdd(kAtomicDelta);
+      b = a + kAtomicDelta - 1;
+      nextIntMax.put(threadIdx, b);
+    }
+
+    nextInt.put(threadIdx, a + 1);
+    return a;
+  }
```

Comment:
These could become methods on the per-thread-id static inner class.
Also comments needed -- it is not obvious initially that we're "allocating" a batch of unique ids from a shared id space using atomics and then handing them out without accessing the atomic.